clang/include/clang/Basic/LangOptions.h

```diff
@@ -623,6 +623,10 @@ class LangOptions : public LangOptionsBase {
   // WebAssembly target.
   bool NoWasmOpt = false;
 
+  /// The default atomic codegen options specified by command line in the
+  /// format of key:{on|off}.
+  std::vector<std::string> AtomicOptionsAsWritten;
```

Comment:
If these are key/value pairs, wouldn't it make more sense to store in a set as opposed to making the consumer of language options deal with parsing?

Reply:
The option value is comma separated key/value pairs delimited by colon whereas the value could be on/off/true/false/yes/no/0/1. Currently the marshaling mechanism of clang option tablegen does not support that. That's why it has to be marshaled through MarshallingInfoStringVector. As a result, it needs to be represented as  std::vector<std::string> and parsed manually.